soroban-env-host/src/events/internal.rs

```diff
@@ -130,14 +95,27 @@ impl InternalEventsBuffer {
 
     /// Converts the internal events into their external representation. This should only be called
     /// either when the host is finished (via `try_finish`), or when an error occurs.
-    // Metering: the new vec allocation is not charged, but that should be fine.
+    // TODO: Metering for non-diagnostic events?
     pub fn externalize(&self, host: &Host) -> Result<Events, HostError> {
         let vec: Result<Vec<HostEvent>, HostError> = self
             .vec
             .iter()
-            .map(|e| match e {
-                InternalEvent::Contract(c) => Ok(HostEvent::Contract(c.clone().to_xdr(host)?)),
-                InternalEvent::Debug(d) => Ok(HostEvent::Debug(d.clone())),
+            .filter(|e| (host.is_debug() || !matches!(e.0, InternalEvent::Contract(_)) || !e.1)) //filter out rolledback Contract events if diagnostics are disabled
```

Comment:
This filter condition seems odd to me -- it's both hard to read / make sense of and .. I'm not really sure it's right? Like do we want to change the contract events we report depending on the diagnostic-enabled flag? I think that's wrong since it'll change the hash. Unless the "filtered out bit" is examined higher up before forming the hash, but in that case .. why filter here additionally? I feel like the "filtered out bit" should probably only be examined in one place, no? (Happy to discuss, this seems like a subtle bit)

Reply:
Yeah good point. stellar-core does all the filtering, so it's not required here. Removed.